src/test/librbd/test_mock_ManagedLock.cc

```diff
@@ -393,6 +399,26 @@ TEST_F(TestMockManagedLock, AcquireLockBlocklist) {
   ASSERT_EQ(0, when_shut_down(managed_lock));
 }
 
+TEST_F(TestMockManagedLock, AcquireLockBlocklistedWatch) {
+  librbd::ImageCtx *ictx;
+  ASSERT_EQ(0, open_image(m_image_name, &ictx));
+
+  MockManagedLockImageCtx mock_image_ctx(*ictx);
+  MockMockManagedLock managed_lock(ictx->md_ctx, *ictx->asio_engine,
```

Comment:
No need for `MockMockManagedLock` since neither of the handler callbacks (`post_reacquire_lock_handler`, etc) is involved here:
```suggestion
  MockManagedLock managed_lock(ictx->md_ctx, *ictx->asio_engine,
```

Reply:
Nice catch, done!